src/test/java/sirius/db/mongo/properties/MongoMultiLanguageStringPropertySpec.groovy

```diff
@@ -177,4 +177,60 @@ class MongoMultiLanguageStringPropertySpec extends BaseSpecification {
         output.getMultiLangTextWithFallback().fetchTextOrFallback() == "OK"
         output.getMultiLangTextWithFallback().getText() == Optional.of("OK")
     }
+
+    def "new null values are not stored"() {
+        given:
+        CallContext.getCurrent().setLang("en")
+        def entity = new MongoMultiLanguageStringEntity()
+        entity.getMultiLangTextWithFallback().addText(null)
+        entity.getMultiLangTextWithFallback().addText("de", "Super")
+        entity.getMultiLangTextWithFallback().addFallback(null)
+        mango.update(entity)
+
+        when:
+        def output = mango.refreshOrFail(entity)
+
+        then:
+        output.getMultiLangTextWithFallback().fetchText() == null
+        output.getMultiLangTextWithFallback().fetchText("en") == null
+        output.getMultiLangTextWithFallback().fetchText("de") == "Super"
+        output.getMultiLangTextWithFallback().fetchText("fallback") == null
+    }
+
+    def "keys with null values are removed from the underlying map if a key already exists"() {
+        given:
+        CallContext.getCurrent().setLang("en")
+        def entity = new MongoMultiLanguageStringEntity()
+        entity.getMultiLangText().addText("en", "Super")
+        mango.update(entity)
+
+        when:
+        entity = mango.refreshOrFail(entity)
+        entity.getMultiLangText().fetchText() == "Super"
```

Comment:
it's a bit confusing to have a condition in a 'when' block and i'm not sure if this will result in a test failure because its not in a then block (or and: chained to a then block)

Reply:
Thx! Fixed now.